File: Java/Medium/CompileError/all-paths-from-source-to-target_1.java
class Solution {
    public List<List<Integer>> allPathsSourceTarget(int[][] graph) {
        
        List<List<Integer>> result = new ArrayList<>();
        List<Integer> path = new ArrayList<>();
        path.add(0);
        dfs(graph, 0, result, path);
        return result;
    }
    
    public void dfs(int[][] graph, int source, List<List<Integer>> result, List<Integer> path) {
        if (source == graph.length - 1) {
            result.add(new ArrayList<>(path));
            return;
        }
        
        for (int i = 0; i < graph[source].length; i++) {
            path.add(graph[source][i]);
            dfs(graph, graph[source][i], res, path);
            path.remove(path.size() - 1);
        }
    }
}

File: C++/Medium/Accepted/all-paths-from-source-to-target_1.cpp
class Solution {
public:
    vector<vector<int>> allPathsSourceTarget(vector<vector<int>>& graph) {
        
        vector<vector<int>> result;
        vector<int> path;
        path.push_back(0);
        dfs(graph, result, path, 0);
        return result;
    }
    void dfs(vector<vector<int>>& graph, vector<vector<int>>& result, vector<int>& path, int node) {
        if (node == graph.size() - 1) {
            result.push_back(path);
            return;
        }
        for (int i = 0; i < graph[node].size(); i++) {
            path.push_back(graph[node][i]);
            dfs(graph, result, path, graph[node][i]);
            path.pop_back();
        }
    }
};